### PR TITLE
Preserve the comments in STM file on subsetting data directory

### PR DESCRIPTION
**Pull request by akshayc11**
Addresses #1765

### CODE REVIEW

File: egs/wsj/s5/utils/subset_data_dir.sh [hunk 1]
```diff
@@ -124,8 +124,10 @@ function do_filtering {
      [ -f $srcdir/reco2file_and_channel ] && \
        utils/filter_scp.pl $destdir/reco <$srcdir/reco2file_and_channel >$destdir/reco2file_and_channel
 
-     # Filter the STM file for proper sclite scoring (this will also remove the comments lines)
-     [ -f $srcdir/stm ] && utils/filter_scp.pl $destdir/reco < $srcdir/stm > $destdir/stm
+     # Filter the STM file for proper sclite scoring
+     # Copy over the comments from STM file
+     [-f $srcdir/stm ] && grep ";;" $srcdir/stm > ${destdir}/stm
```
[the remark this answers is not in the document]
Comment (akshayc11) [reply egs/wsj/s5/utils/subset_data_dir.sh]: Apologies... fixed...

File: egs/wsj/s5/utils/subset_data_dir.sh [hunk 2]
```diff
@@ -124,8 +124,10 @@ function do_filtering {
      [ -f $srcdir/reco2file_and_channel ] && \
        utils/filter_scp.pl $destdir/reco <$srcdir/reco2file_and_channel >$destdir/reco2file_and_channel
 
-     # Filter the STM file for proper sclite scoring (this will also remove the comments lines)
-     [ -f $srcdir/stm ] && utils/filter_scp.pl $destdir/reco < $srcdir/stm > $destdir/stm
+     # Filter the STM file for proper sclite scoring
+     # Copy over the comments from STM file
+     [ -f $srcdir/stm ] && grep ";;" $srcdir/stm > ${destdir}/stm
```
Comment (danpovey) [inline egs/wsj/s5/utils/subset_data_dir.sh]: I think better to use '^;;' as the pattern, to avoid over-matching.

**Comment by akshayc11** [comment]
@danpovey Addressed your comments.

File: egs/wsj/s5/utils/subset_data_dir.sh [hunk 3]
```diff
@@ -124,8 +124,10 @@ function do_filtering {
      [ -f $srcdir/reco2file_and_channel ] && \
        utils/filter_scp.pl $destdir/reco <$srcdir/reco2file_and_channel >$destdir/reco2file_and_channel
 
-     # Filter the STM file for proper sclite scoring (this will also remove the comments lines)
-     [ -f $srcdir/stm ] && utils/filter_scp.pl $destdir/reco < $srcdir/stm > $destdir/stm
+     # Filter the STM file for proper sclite scoring
+     # Copy over the comments from STM file
+     [ -f $srcdir/stm ] && grep "^;;" $srcdir/stm > ${destdir}/stm
```
Comment (danpovey) [inline egs/wsj/s5/utils/subset_data_dir.sh]: can you remove the curly braces from destdir?  It's just inconsistent-looking.

**Pull request merged**